### DISCUSSION TITLE
Poll: SciTools Code of Conduct

### DISCUSSION THREAD

**Discussion by bjlittle** [SciTools/iris#5197] [Polls]
Please cast your vote for which **Code of Conduct** (CoC) SciTools should adopt for the community.

### Reference

For further details about each option, see below:

1. [Citizen Code of Conduct](https://github.com/stumpsyn/policies/blob/master/citizen_code_of_conduct.md)
2. [Contributor Covenant](https://www.contributor-covenant.org/)
3. [NumPy Code of Conduct](https://numpy.org/code-of-conduct/)`†`
4. [SciPy Code of Conduct](https://docs.scipy.org/doc/scipy/dev/conduct/code_of_conduct.html)`†`
5. [Napari Code of Conduct](https://napari.org/stable/community/code_of_conduct.html)`†` `‡`
6. `Other` - This option covers any other CoC
7. `None` - We don't need a CoC

`† - Or a CoC variant derived from this`
`‡ - Example of a CoC derived from [3] & [4]`

Also see this [GH Discussion](https://github.com/SciTools/iris/discussions/5196) for further details.

### Warning

⚠️ If a new CoC option is added to the poll, then all votes require to be re-cast!

### Deadline

🕔 The vote is open until **17h00 (GMT), Fri 28 Apr 23**.

**Comment by pp-mo** [comment]
I note that the numpy CoC references a debt to the SciPy one (right at the end), and the napari one references both of those.
Actually, by direct comparision, the actual differences are pretty tiny !!

Does this mean we should amalgamate any votes for these into 1 category ??

**Reply by bjlittle** [reply]
For me `[3]`, `[4]`, and `[5]` are shades of grey, but my take on it was that if you vote for `[3]` or `[4]` you want specifically that CoC, whereas a vote for `[5]` is an amalgamation/customization of `[3]` and `[4]` i.e., `[5]` is a specific example of that.

Does that help?

**Comment by lbdreyer** [comment]
Why is this code of conduct limited to Iris? Is there a benefit to it being specific to a repo? I would prefer we adopt it across the SciTools community.

A somewhat related question I had was about how we would go about banning someone were that to be necessary (all code of conducts mention banning in some form). I see github has a way to block a user from an organisation (https://docs.github.com/en/communities/maintaining-your-safety-on-github/blocking-a-user-from-your-organization)

We also need to agree on where people can report incidents (again all code of conducts mention this in some form). We can use the email scitools.pub@gmail.com but we also need to agree for this to be monitored.

**Reply by bjlittle** [reply]
@lbdreyer For me this CoC is for all repos across SciTools, not just iris 👍

**Reply by trexfeathers** [reply]
OK I'll change the title

**Reply by trexfeathers** [reply]
Note I can't change the poll title without invalidating it

**Reply by bjlittle** [reply]
TBH the result is the result. We can easily discuss the implications afterwards e.g., the scope of adoption etc. I'm okay with that, without invalidating this poll.

If I had to wait another month for another decision then personally I'd just give up.

The win for me is that everyone's head (that cares) is in the space. We can simply roll from there ...

**Reply by lbdreyer** [reply]
In which case it is probably also a good idea to ping @Scitools/developers as this discussion has been raised in the Iris space and some may not follow iris, but still contribute to SciTools

**Reply by lbdreyer** [reply]
Just found out that there are organisation level discussions (https://docs.github.com/en/organizations/managing-organization-settings/enabling-or-disabling-github-discussions-for-an-organization) But it is implemented in an odd way; it requires a "source repository" and then all discussions for that repo will appear at the org level _and_ in the source repo. So we wouldn't want to use the iris repo as then all the Q/A discussions would also be at org level. But we could consider creating a repo dedicated to org level discussions

I mention this as I suspect these organisation level discussions are going to come up frequently. For example, the ruff discussion(#5254) is also about applying across Scitools but was raised in the Iris space
